src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp

```diff
@@ -156,6 +159,7 @@ namespace Quantum
             typedef unsigned (*TInit)();
             static TInit initSimulatorInstance = reinterpret_cast<TInit>(this->GetProc("init"));
 
+            qubitManager = new CQubitManager();

```

Comment:
If you make this member variable an `std::unique_ptr<CQubitManager>` then this line can be `qubitManager = std::make_unique<CQubitManager>();` and you can remove the explicit delete in the destructor below.

Reply:
Makes sense. Automatic memory management is a good idea when it is applicable.